### PR TITLE
Make the demo a bit clearer {READY_FOR_REVIEW}

### PR DESCRIPTION
**Pull request by justinaray**
* Convert url/programmatic nav into an exclusive state with a toggle
    * Add active top-level nav styling to make routing more obvious
    * Add labels in component html to make the examples a bit easier to grok
    * Default to the first defined route on initial load, if needed
* Dynamically generate nav menus and register routes from the same nav definition
* Styled Header, nav, routing option toggle, and publish button
* Removed state and helpers from window as the listeners are attached in js
* Demo js code organization and comments

### CODE REVIEW

File: js/demo-app/public/index.html [hunk 1]
```diff
@@ -13,20 +13,27 @@
...
-            <span>Programmatically:</span>
-            <br/>
-            <button class="nav-item" onclick="setRoute('/one')">Component 1</button>
-            <button class="nav-item" onclick="setRoute('/two')">Component 2</button>
-        </div>
+        <header>
+            <div class="title">
+                IFrame Coordinator Demo
+            </div>
+            <nav>
```
[the remark this answers is not in the document]
Comment (justinaray) [reply js/demo-app/public/index.html]: @j2pichon ... I did want to add active styling for the programmatic routes but ran out of time.  Also, I didn't know if I should style them as tabs or use another methaphor.  Maybe I can pick your brain on that if time allows tomorrow.

File: js/demo-app/src/index.js [hunk 2]
```diff
@@ -40,43 +72,30 @@ document.getElementById("publish").addEventListener("click", event => {
   });
 });
 
-// Routing helpers
```
Comment (justinaray) [inline js/demo-app/src/index.js]: Just code organization.  Moved 👇

File: js/demo-app/src/index.js [hunk 3]
```diff
@@ -4,22 +4,54 @@ import coordinator from "iframe-coordinator/host.js";
...
+  },
+  {
+    id: 'client2',
+    title: 'Component 2',
+    url: new URL("/components/example2/", window.location).toString(),
+    assignedRoute: "/two",
+    children: ['first', 'second']
+  },
+  {
+    id: 'wikipedia',
```
Comment (justinaray) [inline js/demo-app/src/index.js]: Even though it doesn't participate in routing, I thought this helps to show users we really are dealing with iframes and embeddable content.  It also can help make sure we're handling fq paths in the routing correctly.

If there's any opposition to this, easy to pull out.  Not tied to it.

File: js/demo-app/public/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (j2pichon) [reply js/demo-app/public/index.html]: Sure, as soon as it's merged i'll check that out :)

**Pull request merged**